.drone.yml

```diff
@@ -407,6 +407,72 @@ steps:
     depends_on:
       - build
 
+---
+kind: pipeline
+type: docker
+name: testing-e2e
+
+platform:
+  os: linux
+  arch: amd64
+
+#depends_on:
+#  - testing-amd64
+#  - testing-arm64
+
+trigger:
+  event:
+    - push
+    - tag
+    - pull_request
+
+services:
+  - name: pgsql
+    pull: default
+    image: postgres:10
+    environment:
+      POSTGRES_DB: test
+      POSTGRES_PASSWORD: postgres
+
+steps:
+  - name: fetch-tags
+    image: docker:git
+    commands:
+      - git fetch --tags --force
+    when:
+      event:
+        exclude:
+          - pull_request
+
+  - name: tag-pre-condition
+    pull: always
+    image: drone/git
+    commands:
+      - git update-ref refs/heads/tag_test ${DRONE_COMMIT_SHA}
+
+  - name: build
+    pull: always
+    image: gitea/test_env:linux-amd64  # https://gitea.com/gitea/test-env
+    commands:
+      - curl -sL https://deb.nodesource.com/setup_16.x | bash - && apt-get install -y nodejs
```

Comment:
A little off topic, it's better for the owners to maintain a latest test_env to get nodejs , then we do not need to do `curl` again